test/e2e/networking_perf.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+// Tests network performance using iperf or other containers.
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/intstr"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Declared as Flakey since it has not been proven to run in parallel on small nodes or slow networks in CI
+var _ = Describe("Networking IPerf [Experimental] [Networking] [Flaky]", func() {
+	gceBandwidthBits := int64(30000000000)
+
+	f := NewDefaultFramework("network-perf")
+	run(f, 1, 1, gceBandwidthBits)
+	run(f, 1, 5, gceBandwidthBits)
+	run(f, 1, 8, gceBandwidthBits)
+	// as many clients as possible based on node count.
+	run(f, 1, 10000, gceBandwidthBits)
+})
+
+func run(f *Framework, numServer int, numClient int, maxBandwidthBits int64) {
```

Comment:
Given that there may be multiple tests in this file you may want to call this something less generic.